ddl/index.go

```diff
@@ -680,29 +670,34 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 	return ver, errors.Trace(err)
 }
 
-// checkAndInitLightningBackfillCtx determines which backfill process will be used. The following three processes are available:
-//  1. Txn backfill(original): all the index KVs are written through the transaction interface.
-//  2. Lightning backfill: the index KVs are encoded to SST files and imported to the storage directly.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-//  3. Txn-merge backfill: the backfill index KVs are written through the transaction interface.
-//     The incremental index KVs written by DML are redirected to a temporary index.
-//     After the backfill is finished, the temporary index records are merged back to the original index.
-func checkAndInitLightningBackfillCtx(ctx context.Context, job *model.Job, indexInfo *model.IndexInfo) {
+// pickBackfillProcess determines which backfill process will be used.
+func pickBackfillProcess(w *worker, job *model.Job, indexInfo *model.IndexInfo) model.ReorgType {
+	if job.ReorgMeta.Started {
+		// The backfill task has been started.
+		// Don't switch the backfill process.
+		return job.ReorgMeta.ReorgTp
+	}
 	if IsEnableFastReorg() {
-		if lightning.GlobalEnv.IsInited {
-			err := lightning.BackCtxMgr.Register(ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
+		piTREnabled := isPiTREnable(w)
+		if lightning.GlobalEnv.IsInited && !piTREnabled {
+			err := lightning.BackCtxMgr.Register(w.ctx, indexInfo.Unique, job.ID, job.ReorgMeta.SQLMode)
 			if err != nil {
-				return // Txn backfill
+				logutil.BgLogger().Info("fallback to default backfill process", zap.Error(err))
+				job.ReorgMeta.ReorgTp = model.ReorgTypeTxn
```

Comment:
Once IsEnableFastReorg， even backend is not create successful, it could goto ReorgTypeTxnMerge